MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,778 @@
+Please make sure to remove any explicit bindings of `ModelSource` in your DI configuration.
+
+For the standalone use case this means that the `GLSPDiagramServer` model source implementation is no longer available.
+Explicit bindings of this service identifier can be simply removed.
+In 1.0.0 the `GLSPDiagramServer` also served as explicit `ActionHandler`.
+This behavior is discouraged in 2.x.
+If you have been using a custom diagram server that handles additional actions please migrate the handling to an explicit action handler and configure in your diagram module (`configureActionHandler(...)`).
+
+### Introduction of `FeatureModule`s
+
+With 2.x all base diagram feature modules have been refactored and now extend a custom `FeatureModule` class instead of the default inversify `ContainerModule`.
+`FeatureModule`s are special `ContainerModule`s that also bind a unique service identifier when loaded into the container.
+This allows runtime checks to see if a module is configured (i.e. loaded into the container) or not.
+In addition,this enables the declaration of dependency chains i.e. application-context specific modules that add additional functionality on top of a core feature module.
+For instance let's have a look at the `theiaSelectModule`
+
+```ts
+export const theiaSelectModule = new FeatureModule(
+    bind => {
+        bindAsService({ bind }, TYPES.ISelectionListener, TheiaGLSPSelectionForwarder);
+    },
+    { requires: selectModule }
+);
+```
+
+This module requires the `selectFeature` module.
+This means it will only be loaded into the container if the `selectModule` has been loaded before.
+
+Depending on your set of custom modules some migration effort might be necessary.
+Custom modules that are loaded in addition to core GLSP modules should work as before in 2.x without any changes.
+Custom modules that are a fullfledged replacement for a GLSP core feature module need to be migrated to a `FeatureModule`.
+To keep the dependency chain intact the replacement module has to reuse the `featureId` of the base module it is replacing.
+For instance if you have a custom `myViewportModule` that serves as a replacement for the base `viewportModule` than the module
+declaration should look like this:
+
+```ts
+export const myViewportModule = new FeatureModule(
+    (bind, _unbind, isBound) => {
+        /// your custom module configuration
+    },
+    { featureId: viewportModule.featureId }
+);
+```
+
+In addition, the naming of feature modules in 1.0.0 was an inconsistent mix of modules with no prefix, prefixed with `glsp` and prefixed with `sprotty`.
+In 2.x the naming has been aligned and all feature modules provided by GLSP now have no prefix.
+In addition, several modules have been renamed to better reflect the purpose of the module.
+We recommend to use `Search & Replace` to migrate affected module references.
+
+<details>
+  <summary>List of changes</summary>
+
+-   defaultGlspModule-> baseModule
```

Comment:
If we code quote the renames above, these should also be code quoted, i.e., `defaultGlspModule`.